src/main/java/mekanism/common/content/gear/mekatool/ModuleVeinMiningUnit.java

```diff
@@ -49,34 +59,76 @@ public int getExcavationRange() {
         return excavationRange.get().getRange();
     }
 
-    public static Set<BlockPos> findPositions(BlockState state, BlockPos location, Level world, int maxRange) {
-        Set<BlockPos> found = new LinkedHashSet<>();
-        Set<BlockPos> openSet = new LinkedHashSet<>();
-        openSet.add(location);
-        Block startBlock = state.getBlock();
-        int maxCount = MekanismConfig.gear.disassemblerMiningCount.get() - 1;
-        while (!openSet.isEmpty()) {
-            BlockPos blockPos = openSet.iterator().next();
-            found.add(blockPos);
-            openSet.remove(blockPos);
-            if (found.size() > maxCount) {
-                return found;
+    // Helper class to help calculate the breadth first traversal path distance
+    private static class TraversalDistance {
+
+        // Start at distance 0
+        private int distance = 0;
+        private int next;
+
+        // Initialize with the number of elements at distance 0
+        public TraversalDistance(int next) {
+            this.next = next;
+        }
+
+        // When all elements at distance 0 are found, determine how many elements there are with distance 1 and increment distance
+        public void updateDistance(int found, int frontierSize) {
+            if (found == next) {
+                distance++;
+                next += frontierSize;
+            }
+        }
+
+        public int getDistance() {
+            return distance;
+        }
+    }
+
+    public static boolean canVeinBlock(BlockState state) {
+        //Even though we now handle breaking bounding blocks properly, don't allow vein mining them
+        return !(state.getBlock() instanceof BlockBounding);
+    }
+
+    public static Object2IntMap<BlockPos> findPositions(Level world, Map<BlockPos, BlockState> initial, int extendedRange, Object2BooleanMap<Block> oreTracker) {
+        Object2IntMap<BlockPos> found = new Object2IntLinkedOpenHashMap<>();
+
+        int maxVein = MekanismConfig.gear.disassemblerMiningCount.get();
+        int maxCount = initial.size() + maxVein * oreTracker.size();
+
+        Map<BlockPos, BlockState> frontier = new LinkedHashMap<>(initial);
+        TraversalDistance dist = new TraversalDistance(frontier.size());
+        while (!frontier.isEmpty()) {
+            Iterator<Entry<BlockPos, BlockState>> iterator = frontier.entrySet().iterator();
+            Entry<BlockPos, BlockState> blockEntry = iterator.next();
+            iterator.remove();
+
+            BlockPos blockPos = blockEntry.getKey();
+            BlockState blockState = blockEntry.getValue();
```

Comment:
No real reason to initialize a local block state variable until after the found size vs max count is checked. In fact as we only use it to get the block in one spot we may as well have it be `Block block = blockEntry.getValue().getBlock();`